src/Illuminate/Contracts/Cache/Repository.php

```diff
@@ -38,7 +38,7 @@ public function pull($key, $default = null);
      *
      * @param  string  $key
      * @param  mixed   $value
-     * @param  \DateTimeInterface|\DateInterval|float|int  $minutes
+     * @param  \DateTimeInterface|\DateInterval|float|int|null  $minutes
```

Comment:
Perhaps this is off-topic/not the right place for this conversation (feel free to tell me to open a new issue)...

I'm thinking that the perhaps solution here should be the other way round: `null` should not be allowed. It doesn't make sense to pass `null` here, because then `Cache::put('foo', 'bar');` (which is possible when you have a default value of `null` such as in the `\Illuminate\Cache\Repository::put` implementation) is completely a no-op due to the `!is_null(...)` check inside that implementation. I think a better solution would be to forbid `null` being passed (so `\DateTimeInterface|\DateInterval|float|int` would be correct), and remove the `$minutes = null` from the implementations, thus not allowing something that would be pointless. Perhaps `0` is a valid value, but that I think is a separate discussion (typically, I'd interpret that as "cache this forever", but as I said - separate discussion)

Reply:
@asgrim heya James, we talked about the `! is_null` check here as well: https://github.com/laravel/framework/issues/27160#issuecomment-454019364

Marco pointed out that this is indeed a problem because this check doesn't allows you to implement a default on the driver itself. I'm currently working on another PR to remove this check so driver could implement a default if they wish.

Reply:
Regarding "cache this forever": currently we're following the PSR interface which I don't think allows for this?

Reply:
Oh, yes, sorry -I'd read that, but I neglected to mention the problem with that - `\Illuminate\Contracts\Cache\Store::put` interface does not allow `null` either, so storage implementations should not accept `null` either:

https://github.com/laravel/framework/blob/1f562c8b19e2edf1144348937d1f121559257a85/src/Illuminate/Contracts/Cache/Store.php#L25-L33

Reply:
> Regarding "cache this forever": currently we're following the PSR interface which I don't think allows for this?

It's actually the opposite (assuming PSR-16 Simple Cache here):

> If a calling library asks for an item to be saved but does not specify an expiration time, or specifies a null expiration time or TTL, an Implementing Library MAY use a configured default duration. If no default duration has been set, the Implementing Library MUST interpret that as a request to cache the item forever, or for as long as the underlying implementation supports.

And therein lies the reason why the `null` handling here seems backwards to me: I would expect `null` to be "cache forever" or "cache for default (configured?) duration". The current implementation is "do not cache, do not notify user, do not pass go" :grin:

Reply:
Note: I checked, same wording used in PSR-6 too.

Reply:
Ah, very good point indeed. Thanks for pointing that out. This indeed also needs to be adjusted.

On second thought, it might be better if I send in the changes together in one PR.